src/note_encryption/orchard_domain.rs

```diff
@@ -0,0 +1,61 @@
+use core::fmt;
+
+use zcash_note_encryption_zsa::{AEAD_TAG_SIZE, MEMO_SIZE};
+
+use crate::{action::Action, note::Nullifier, Note};
+
+use super::{
+    note_bytes::{NoteByteReader, NoteByteWriter},
+    Memo,
+};
+
+/// Represents the Orchard protocol domain specifics required for note encryption and decryption.
+pub trait OrchardDomain: fmt::Debug + Clone {
```

Comment:
From @PaulLaux:
let's rename to EncDomain

Reply:
In our discussion, we decided to keep the name `OrchardDomain` for now (along with `OrchardDomainBase`, `OrchardDomainVanilla`, and `OrchardDomainZSA`) because we have three traits - `OrchardDomain`, `OrchardCircuit`, and `OrchardHash`. This naming scheme with `Orchard` prefix looks consistent.